doc/continuous_zooming.md

```diff
@@ -0,0 +1,89 @@
+### Implementation approach and algorithms
+
+While the communication scheme subtlely explained above is fine for the regular functioning mode, it presents some drawbacks for the CZ mode.
+
+In the regular mode, every time a new fractal image has to be drawn, C++ performs some boilerplate initializations. Doing this for an on-going animation that creates a new image every few miliseconds is unbearable.
+
+For that reason, scheme communication for CZ is slightly different. Instead of using the _regular_ image calculation trigger API from the `fract4dc` extension, a new API has been created with the following conceptual facade:
+- start CZ process: this triggers the image calculation process and perform boilerplate initializations on the C++ layer. It also creates an infinite loop waiting for updates from the python layer.
+- update CZ location: sends an update to be processed by the infinite loop from the previous API.
+- stop CZ process: tells the infinite loop to stop waiting for updates and finish off.
+
+The other communication flow remains the same: python part will be reading the pipe to draw image updates from C++.
+#### Pixel reutilization algorithm
+
+This is the main algorithm and the foundation of the CZ feature.
+The basic idea behind this is that, for every new frame you want to draw there are some pixels from the previous frame you can reuse.
+
+While Gnofract 4D utilizes some algorithms to improve performance, the basic idea behind the image calculation process is that every single pixel of the image is assigned a point in the complex plane XY (actually Gnofract allows to work in a 4D space with six plane rotation), and that point is the _driving_ parameter to pass to the fractal formula to calculate every pixel color.
+
+`Location` (`X`, `Y` ... and `Z`, `W` in 4D space) and `Size` parameters will determine which point of the complex plane correspond to every pixel of the image (we leave out from this explanation the plane rotation parameters). This makes very likely to happen that subsequent frames, where location and size would change slightly, have pixels with the same, or very close point in the complex plane. So it makes sense to, instead of calculate the color of that pixel again (which is the heaviest process) we take the already calculated value.
+
+The process to find those pixels with same or close complex plane point values, in an efficient way, is based in the assumption that _imaginary values `y` remain constant while moving across X axis, while real values `x` do the same for the Y axis_. And it consist in the following steps:
+- find all the `x` values or columns for the new fractal (image) that are close enough to the previous image ones.
```

Comment:
One way of thinking about this: imagine you are taking the previous frame and scaling it up. In frame 1, to get the color for pixel (0,0) we calculated point (x,y,z,w). In Frame 2, point (x,y,z,w) is now at a different pixel location, but if that location is still on screen, we can reuse the calculation and color pixel(a,b) with that point. I'm pretty sure that doesn't really depend on only x changing as we move across the screen or y changing vertically. We have a vector (dx,dy,dz,dw) which is the amount the point changes when we step 1 pixel to the left - we can use that to generalize this to work for rotated images or ones in other planes.

Reply:
That was exactly my thought when I first read https://github.com/xaos-project/XaoS/wiki/Developer's-Guide#saving-previous-pixels.
Then I said to myself "ok, let's try only XY plane (no plane rotation)". I was surprised when I found out you can actually change plane rotations and it keeps working. Except in certain cases.
As the Gnofract 4D manual states:
> It allows you to treat a fractal which has more than one parameter as a four-dimensional object and interactively view slices of this object from arbitrary angles, giving rise to some very unusual images.
It seems that, as long as the slice you are viewing is perpendicular or parallel to the XY plane the assumption is valid. In other words, if you rotate the slice respect to X and Y at the same time, the it  doesn't work. 
I tried to solve this issue/limitation, being playing around with the rotation matrix, point vectors... but honestly I don't have the math background to.